src/transformStateWithClones.js

```diff
@@ -7,7 +7,22 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let newState = { ...state };
+  const states = [];
+
+  for (const key of actions) {
```

Comment:
The variable `key` is used to iterate over `actions`, which implies that `actions` is an array of objects. Ensure that each object in `actions` has a `type` property and the necessary properties for each action type.